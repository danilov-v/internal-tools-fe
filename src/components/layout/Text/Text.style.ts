import styled from 'styled-components';

type ContainerProps = {
  align: 'center' | 'justify' | 'left' | 'right';
  mb: number;
  mt: number;
  variant: 'error' | 'primary' | 'secondary';
};

const VARIANTS = {
  error: {
    color: 'error',
    fontSize: 14,
  },
  primary: {
    color: 'primaryGray',
    fontSize: 18,
  },
  secondary: {
    color: 'lightGray',
    fontSize: 16,
  },
};

function getVariantProperty(
  variant: 'error' | 'primary' | 'secondary',
  property: 'color',
): 'error' | 'primaryGray' | 'lightGray';

function getVariantProperty(
  variant: 'error' | 'primary' | 'secondary',
  property: 'fontSize',
): number;

function getVariantProperty(
  variant: 'error' | 'primary' | 'secondary',
  property: 'color' | 'fontSize',
): string | number {
  return VARIANTS[variant][property];
}

export const Container = styled.p<ContainerProps>`
  color: ${({ theme, variant }) => theme[getVariantProperty(variant, 'color')]};
  font-size: ${({ variant }) => getVariantProperty(variant, 'fontSize')}px;
  margin-bottom: ${({ mb }) => mb}px;
  margin-top: ${({ mt }) => mt}px;
  text-align: ${({ align }) => align};
`;
