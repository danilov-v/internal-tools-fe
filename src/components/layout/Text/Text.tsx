import React from 'react';

import * as S from './Text.style';

const Text: React.FC<TextProps> = ({
  align = 'left',
  children = '',
  className = '',
  component = 'p',
  mb = 0,
  mt = 0,
  variant = 'primary',
}) => (
  <S.Container
    align={align || 'left'}
    as={component}
    className={className}
    mb={mb}
    mt={mt}
    variant={variant || 'primary'}
  >
    {children}
  </S.Container>
);

type TextProps = {
  align?: 'center' | 'justify' | 'left' | 'right';
  className?: string;
  component?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';
  mb?: number;
  mt?: number;
  variant?: 'error' | 'primary' | 'secondary';
};

export { Text };
