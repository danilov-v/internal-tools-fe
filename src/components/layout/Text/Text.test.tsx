import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import 'jest-styled-components';

import { Text } from './Text';

const getComponent = (props = {}): RenderResult => {
  const parsedProps = {
    children: '',
    ...props,
  };

  return render(<Text {...parsedProps} />);
};

describe('Text component', () => {
  it('matches snapshot with only required props', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });

  it('matches snapshot with additional props', () => {
    const primaryText = getComponent({
      align: 'center',
      children: 'children',
      mb: 100,
      mt: 20,
      variant: 'primary',
    });

    expect(primaryText.baseElement).toMatchSnapshot();
  });
});
