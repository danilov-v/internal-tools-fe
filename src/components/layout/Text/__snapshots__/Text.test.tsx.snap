// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component matches snapshot with additional props 1`] = `
<body>
  .c0 {
  font-size: 18px;
  margin-bottom: 100px;
  margin-top: 20px;
  text-align: center;
}

<div>
    <p
      class="c0"
    >
      children
    </p>
  </div>
</body>
`;

exports[`Text component matches snapshot with only required props 1`] = `
.c0 {
  font-size: 18px;
  margin-bottom: 0px;
  margin-top: 0px;
  text-align: left;
}

<body>
  <div>
    <p
      class="c0"
    />
  </div>
</body>
`;
